### PR TITLE
Added UI to Area dropdown and Iteration dropdown

### PR DESCRIPTION
**Pull request by SMahil**
- Added updated UI to area and Iteration dropdown
- Fix Overlap Issue among same
- Added left side truncation to area and iteration list

Closes: #1273 #1443 #993 #1370

### CODE REVIEW

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 1]
```diff
@@ -195,71 +195,88 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
+            <input type="hidden" name="" value="">
+            <span *ngIf="!searchArea" (click)="activeSearchArea()"
+               [class.none-assigned]="!workItem.relationships?.area?.data?.attributes?.name" class="pointer" id="WI_details_area">
+               {{ workItem.relationships?.area?.data?.attributes?.name || "None" }}
+            </span>
+            <div class="flex-container">
+              <div *ngIf="searchArea" class=" col-sm-6 padding-0">
+                <div class="input-group " dropdown>
+                  <input id="areaSearchInput" #areaSearch (keyup)="filterArea($event)" placeholder="Select Area"
+                    autocomplete="off" value="{{ workItem.relationships?.area?.data?.attributes?.name }}"
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: Remove value="{{ workItem.relationships?.area?.data?.attributes?.name }}"

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 2]
```diff
@@ -195,71 +195,88 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
+                      *ngFor="let area of filteredAreas;"
+                      [attr.data-value]="area.id"
+                      (click)="showAreaOnInput(area)" [id]="area.id">
+                      <a class="truncate left-truncate">
+                        <span [innerHTML]="area.attributes?.name | almSearchHighlight: areaSearch.value" >
+                        </span>
+                      </a>
+                    </li>
+                  </ul>
+                  <span class="input-group-addon dropdown-toggle" dropdownToggle data-dropdown="dropdown">
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: Add #areaClick

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 3]
```diff
@@ -196,70 +196,85 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
+            <input type="hidden" name="" value="">
+            <span *ngIf="!searchArea" (click)="activeSearchArea()"
+               [class.none-assigned]="!workItem.relationships?.area?.data?.attributes?.name" class="pointer" id="WI_details_area">
+               {{ workItem.relationships?.area?.data?.attributes?.name || "None" }}
+            </span>
+            <div class="flex-container">
+              <div *ngIf="searchArea" class=" col-sm-6 padding-0">
+                <div class="input-group " dropdown>
+                  <input id="areaSearchInput" #areaSearch (keyup)="filterArea($event)" placeholder="Select Area"
+                    autocomplete="off" value="{{ workItem.relationships?.area?.data?.attributes?.name }}"
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: Remove value="{{ workItem.relationships?.area?.data?.attributes?.name }}"

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 4]
```diff
@@ -196,70 +196,85 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
+                      *ngFor="let area of filteredAreas;"
+                      [attr.data-value]="area.id"
+                      (click)="showAreaOnInput(area)" [id]="area.id">
+                      <a class="truncate left-truncate">
+                        <span [innerHTML]="area.attributes?.name | almSearchHighlight: areaSearch.value" >
+                        </span>
+                      </a>
+                    </li>
+                  </ul>
+                  <span class="input-group-addon dropdown-toggle" dropdownToggle data-dropdown="dropdown">
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: Make area click functional - 
Step 2 - #areaClick to span tag

File: src/app/work-item/work-item-detail/work-item-detail.component.html (hunk 3 above)
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: To make area selection functional
1. Remove value="{{ workItem.relationships?.area?.data?.attributes?.name }}"

File: src/app/work-item/work-item-detail/work-item-detail.component.html (hunk 4 above)
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: To make area selection functional
2. Add #areaClick to the span tag

File: src/app/work-item/work-item-detail/work-item-detail.component.ts [hunk 5]
```diff
@@ -740,8 +740,8 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
 
   selectIteration(iteration: any): void {
     this.selectedIteration = iteration;
-    this.dropdownButton.nativeElement.innerHTML = this.selectedIteration.attributes.name +
-                                                  ' <span class="caret"></span>'
+   // this.dropdownButton.nativeElement.innerHTML = this.selectedIteration.attributes.name +
+     //                                             ' <span class="caret"></span>'
   }
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: To make area selection functional
(I'm not able to add the comment at the correct line number)
3. Add @ViewChild('areaClick') areaClick: any; before @ViewChild('areaSearch') areaSearch: any;
4. In function activeSearchArea, add this.areaClick.nativeElement.click(); after this.areaSearch.nativeElement.focus();

**Comment by naina-verma** [comment]
LGTM!

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 6]
```diff
@@ -198,73 +198,85 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
-                {{ workItem.relationships?.area?.data?.attributes?.name || "None" }}
-              </div>
-              <div style="display: flex;">
-                <div *ngIf="searchArea" class="area-search">
-                  <div class="list-container">
-                    <span class="fa fa-search area-search-icon"> </span>
-                    <input id="areaSearchInput" #areaSearch (keyup)="filterArea($event)"
-                      autocomplete="off" />
-                    <ul class="item-ul area-list" #areaList>
-                      <li *ngIf="!areas.length" class="textCenter"><i class="fa fa-spinner fa-pulse"></i></li>
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: This logic needs to be included.

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 7]
```diff
@@ -198,73 +198,85 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
-                      <li *ngIf="!filteredAreas.length">No matches found.</li>
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: This logic needs to be included.

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 8]
```diff
@@ -198,73 +198,85 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
-               {{ workItem.relationships?.iteration?.data?.attributes?.name || "None" }}
-            </span>
-            <span *ngIf="searchIteration"
-              id="WI_details_iteration" dropdown>
-              <button class="btn btn-default" type="button" #dropdownButton
-                id="WI_details_iteration_dropdownbtn" data-toggle="dropdown" dropdownToggle>
-                {{ workItem.relationalData?.iteration?.attributes.name|| "None" }} <span class="caret"></span>
-              </button>
-              <ul role="menu" aria-labelledby="WI_details_iteration_dropdown" dropdownMenu>
-                <li *ngIf="!iterations.length" class="textCenter"><i class="fa fa-spinner fa-pulse"></i></li>
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: This logic needs to be included.

File: src/assets/stylesheets/shared/_custom-mixins.scss [hunk 9]
```diff
@@ -97,3 +132,6 @@
     right: 0;
     border-width: 0;
 }
+@mixin none-assigned {
```
Comment (mindreeper2420) [inline src/assets/stylesheets/shared/_custom-mixins.scss]: why add a mixin for just a color?

File: src/assets/stylesheets/shared/main.scss [hunk 10]
```diff
@@ -49,6 +49,10 @@ body {
   position: relative;
 }
 
+.none-assigned {
```
Comment (mindreeper2420) [inline src/assets/stylesheets/shared/main.scss]: is the unassigned used anywhere other than the detail panel? can it just be kept to that component?

File: src/app/work-item/work-item-detail/work-item-detail.component.html [hunk 11]
```diff
@@ -198,73 +198,85 @@ <h2 *ngIf="loggedIn && workItem.id" id="wi-detail-title-click" class="detail-tit
...
-                  <button class="btn btn-lg btn-default"  type="button" (click)=closeAreaDropdown()>
-                    <span class="pficon pficon-close" id="close_area" ></span>
-                  </button>
+          <div class="combobox-container col-sm-6 padding-0 detail-area-name">
+            <input type="hidden" name="" value="">
+            <span *ngIf="!searchArea" (click)="activeSearchArea()"
+               [class.none-assigned]="!workItem.relationships?.area?.data?.attributes?.name" class="pointer" id="WI_details_area">
+               {{ workItem.relationships?.area?.data?.attributes?.name || "None" }}
+            </span>
+            <div class="flex-container">
```
Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: is it possible to take this out of a flex container? flex was removed from the detail panel in preparation for resizing using bootstrap

File: src/assets/stylesheets/shared/main.scss (hunk 10 above)
Comment (SMahil) [inline src/assets/stylesheets/shared/main.scss]: its only in detail page. will keep it in that component

File: src/app/work-item/work-item-detail/work-item-detail.component.html (hunk 11 above)
Comment (SMahil) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: flex container is in work item linking and comments.. we have to remove it from all in that case

Comment (mindreeper2420) [inline src/app/work-item/work-item-detail/work-item-detail.component.html]: lets leave that as another PR - we can fold it into our flex work

File: src/assets/stylesheets/shared/main.scss (hunk 10 above)
Comment (mindreeper2420) [inline src/assets/stylesheets/shared/main.scss]: cool, thanks

**Comment by fabric8cd** [comment]
@SMahil snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1530.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1530.int.rdu2c.fabric8.io
